core/src/worker/workflow/machines/child_workflow_state_machine.rs

```diff
@@ -120,6 +132,14 @@ pub(super) struct Cancelled {
     seen_cancelled_event: bool,
 }
 
+fn completion_of_not_abandoned_err() -> WFMachinesError {
+    WFMachinesError::Nondeterminism(
+        "Child workflows which don't have the ABANDON cancellation type cannot complete after \
```

Comment:
Hrmm, I figured `ABANDON` was a parent-close-only thing. Arguably this is a lang bug if they send you a cancel-after-complete of any child/activity regardless of abandon setting. What situation/lang is sending you cancel of completed-yet-abandoned children?

Reply:
Lang shouldn't have to think about that, that's the point of core. And this isn't cancel after-complete, this is complete after cancel, which is outside of SDK control entirely. The activity never sees a cancel, and thus never knows it shouldn't complete.